src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -1603,30 +1603,86 @@ namespace detail
         std::string name,
         std::shared_ptr< Attribute::resource > resource )
     {
-        detail::AttributeWithShape< char > attr =
-            preloadedAttributes.getAttribute< char >( name );
-        if( attr.shape.size() != 2 )
-        {
-            throw std::runtime_error( "[ADIOS2] Expecting 2D ADIOS variable" );
-        }
-        size_t height = attr.shape[ 0 ];
-        size_t width = attr.shape[ 1 ];
+        /*
+         * char_type parameter only for specifying the "template" type.
+         */
+        auto loadFromDatatype =
+            [ &preloadedAttributes, &name, &resource ]( auto char_type ) {
+                using char_t = decltype( char_type );
+                detail::AttributeWithShape< char_t > attr =
+                    preloadedAttributes.getAttribute< char_t >( name );
+                if( attr.shape.size() != 2 )
+                {
+                    throw std::runtime_error(
+                        "[ADIOS2] Expecting 2D ADIOS variable" );
+                }
+                char_t const * loadedData = attr.data;
+                size_t height = attr.shape[ 0 ];
+                size_t width = attr.shape[ 1 ];
 
-        std::vector< std::string > res( height );
-        for( size_t i = 0; i < height; ++i )
+                std::vector< std::string > res( height );
+                if( std::is_signed< char >::value ==
```

Comment:
Is it ok to add a bit more in-code comments for this part? Just to make it easier for our future selves, reading the workaround.

Reply:
Done